scripts/examples/smapiv2.py

```diff
@@ -1,19 +1,22 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import print_function
-import os, sys, time, socket, traceback
+import os, sys, io, time, socket, traceback
 
-log_f = os.fdopen(os.dup(sys.stdout.fileno()), "aw")
+log_f = io.open(os.dup(sys.stdout.fileno()), "w")
 pid = None
 
 def reopenlog(log_file):
     global log_f
     if log_f:
         log_f.close()
     if log_file:
-        log_f = open(log_file, "aw")
+        try:
+            log_f = io.open(log_file, "a")
+        except FilenotFoundError:
+            log_f = io.open(log_file, "w")
```

Comment:
Opening without the append mode may cause the output file (descriptor in this case) to be truncated. The 1:1 conversion would be to use `"a"` everywhere where `"aw"` had do be used with Python2.

The explanation is here:
https://stackoverflow.com/questions/42824474/python-open-built-in-function-difference-between-modes-aw-and-a-or-aw-and
File creation also works in append mode:
```py
$ rm foo;python3 -c 'import io; f = io.open("foo", "a"); f.write("Hi!\n")';cat foo
Hi!
```
```suggestion
        log_f = io.open(log_file, "a")
```

Reply:
@acefei changed from io.open to open ,unnecessary keys were removed , changed result = map(lambda vi: vdi_info(vi), vis) to list result = [vdi_info(vi) for vi in vis] commit db38533188e21b4507994590f24cf2e8440332ce